src/core/plugin/luapi_application.h

```diff
@@ -327,6 +384,263 @@ static int applib_layerAction(lua_State* L) {
+ *              [6] = 828.2946,
+ *              [7] = 875.1697,
+ *              [8] = 806.0,
+ *              [...] = ...,
+ *            },
+ *            ["width"] = 1.4,
+ *            ["color"] = 0xff000000,
+ *            ["fill"] = 0,
+ *            ["tool"] = "pen",
+ *            ["lineStyle"] = "solid"
+ *        })
+ */
+static int applib_addSpline(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    Control* ctrl = plugin->getControl();
+    PageRef const& page = ctrl->getCurrentPage();
+    Layer* layer = page->getSelectedLayer();
+    Stroke* stroke = new Stroke();
+
+    // Discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TTABLE);
+
+    lua_getfield(L, 1, "splines");
+    if (!lua_istable(L, -2))
+        luaL_error(L, "Missing Spline table!");
+    // Push another reference to the table on top of the stack (so we know
+    // where it is, and this function can work for negative, positive and
+    // pseudo indices
+    // stack now contains: -1 => table
+    lua_pushnil(L);
+    // stack now contains: -1 => nil; -2 => table
+    std::vector<double> coordStream;
+    while (lua_next(L, -2)) {
+        // stack now contains: -1 => value; -2 => key; -3 => table
+        double value = lua_tonumber(L, -1);
+        coordStream.push_back(value);
+        // pop value + copy of key, leaving original key
+        lua_pop(L, 1);
+        // stack now contains: -1 => key; -2 => table
+    }
+    // stack now contains: -1 => table (when lua_next returns 0 it pops the key
+    // but does not push anything.)
+    // Pop table
+    lua_pop(L, 1);  // Stack is now the same as it was on entry to this function
+
+    // Check if the list is divisible by 8.
+    // (Due to the nature of quadratic splines, you'll have startX, startY,
+    // ctrl1X, ctrl1Y, ctrl2X, ctrl2Y, endX, and endY repeating in a pattern.)
+    if (coordStream.size() % 8 != 0)
+        luaL_error(L, "Spline table incomplete!");
+
+    // Now take that gigantic list of splines and create SplineSegments out of them.
+    long unsigned int i = 0;
+    while (i < coordStream.size()) {
+        // start, ctrl1, ctrl2, end
+        Point start = Point(coordStream.at(i), coordStream.at(i + 1), Point::NO_PRESSURE);
+        Point ctrl1 = Point(coordStream.at(i + 2), coordStream.at(i + 3), Point::NO_PRESSURE);
+        Point ctrl2 = Point(coordStream.at(i + 4), coordStream.at(i + 5), Point::NO_PRESSURE);
+        Point end = Point(coordStream.at(i + 6), coordStream.at(i + 7), Point::NO_PRESSURE);
+        i += 8;
+        SplineSegment segment = SplineSegment(start, ctrl1, ctrl2, end);
+        std::list<Point> raster = segment.toPointSequence();
+        for (Point point: raster) stroke->addPoint(point);
+        // TODO: (willnilges) Is there a way we can get Pressure with Splines?
+    }
+    // Finish building the Stroke and apply it to the layer.
+    addStrokeHelper(L, stroke, layer);
+    return 1;
```

Comment:
The number of return values is 0, isn't it?
```suggestion
    return 0;
```